src/lib/components/Cytoscape.react.js

```diff
@@ -196,6 +200,81 @@ class Cytoscape extends Component {
         }, SELECT_THRESHOLD);
 
         // /////////////////////////////////////// EVENTS //////////////////////////////////////////
+        const updateTapContextMenu = (newContext) => {
+            this.props.setProps({ tapContextMenu: newContext });
+        }
+        const updateElements = (newElement) => {
+            var updatedElements = this.props.elements
+            updatedElements.push(newElement)
+            this.props.setProps({ elements: updatedElements });
+        }
+        const contextMenuDefaultFunctions = {
+            remove: function (event) {
+                var target = event.target || event.cyTarget;
+                target.remove();
+            },
+            add_node: function (event) {
+                var data = {
+                    group: 'nodes'
+                };
+                var pos = event.position || event.cyPosition;
+                cy.add({
+                    data: data,
+                    position: {
+                        x: pos.x,
+                        y: pos.y
+                    }
+                });
+            },
+            add_edge: function (event) {
+                const selectedNodeIds = selectedNodes.map(node => node.id());
+                if (selectedNodes.length === 2) {
```

Comment:
What if you haven't selected 2 nodes? This will silently do nothing. Can we make it so an `add_edge` item will be disabled unless this condition is met? And similarly for user functions or callback-triggering events, can we allow the user to define the prerequisite conditions?

Reply:
I've tried to disable the context menu if more/less than 2 nodes are selected. This is a bit tricky since the menu items are first generated and every time a node is selected it would have to be regenerated, which is not ideal. I think it could make sense to add a self looping edge if 1 node is selected, and if three nodes or more are selected we could connect all of the nodes to one another. I could see why someone might find that useful. Or we could add an error message if we don't like that behaviour.

Reply:
1/2/error like you have now seems reasonable for now. Might make sense to have this be an `alert` instead of a `console.log` though? To most users a `console.log` will be invisible, a silent failure. Also, is it possible to get here with 0 nodes selected?

Reply:
Yes - I can add a case for 0 nodes!

Reply:
I have changed them to console.error() so they are more apparent in the console and added a case for 0 nodes:)